[BUG] org.opensearch.indices.replication.RemoteStoreReplicationSourceTests.testGetCheckpointMetadataEmpty is failing

**Describe the bug**

The org.opensearch.indices.replication.RemoteStoreReplicationSourceTests.testGetCheckpointMetadataEmpty is failing on `main`, apparently after https://github.com/opensearch-project/OpenSearch/pull/8433

```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=100, name=opensearch[org.opensearch.indices.replication.RemoteStoreReplicationSourceTests][remote_refresh][T#1], state=RUNNABLE, group=TGRP-RemoteStoreReplicationSourceTests]
Caused by: java.lang.AssertionError
	at __randomizedtesting.SeedInfo.seed([92C4ACD8A776127B]:0)
	at org.opensearch.index.seqno.ReplicationTracker.setLatestReplicationCheckpoint(ReplicationTracker.java:1216)
	at org.opensearch.index.shard.IndexShard.onCheckpointPublished(IndexShard.java:1857)
	at org.opensearch.index.shard.RemoteStoreRefreshListener.syncSegments(RemoteStoreRefreshListener.java:195)
	at org.opensearch.index.shard.RemoteStoreRefreshListener.lambda$afterRefresh$0(RemoteStoreRefreshListener.java:183)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:849)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
```

**To Reproduce**
```
./gradlew ':server:test' --tests "org.opensearch.indices.replication.RemoteStoreReplicationSourceTests.testGetCheckpointMetadataEmpty" -Dtests.seed=92C4ACD8A776127B -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=sk-SK -Dtests.timezone=Canada/Yukon -Druntime.java=20
```

**Expected behavior**
The test should always pass

**Plugins**
Standard

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - CI
 - 
**Additional context**
https://build.ci.opensearch.org/job/gradle-check/20172/
https://build.ci.opensearch.org/job/gradle-check/20196/

--- a/server/src/test/java/org/opensearch/indices/replication/RemoteStoreReplicationSourceTests.java
+++ b/server/src/test/java/org/opensearch/indices/replication/RemoteStoreReplicationSourceTests.java
@@ -45,18 +45,16 @@ public class RemoteStoreReplicationSourceTests extends OpenSearchIndexLevelRepli
 
     private Store remoteStore;
 
+    private final Settings settings = Settings.builder()
+        .put(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, true)
+        .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+        .build();
+
     @Override
     public void setUp() throws Exception {
         super.setUp();
 
-        indexShard = newStartedShard(
-            true,
-            Settings.builder()
-                .put(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, true)
-                .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
-                .build(),
-            new InternalEngineFactory()
-        );
+        indexShard = newStartedShard(true, settings, new InternalEngineFactory());
 
         indexDoc(indexShard, "_doc", "1");
         indexDoc(indexShard, "_doc", "2");
@@ -133,7 +131,7 @@ public class RemoteStoreReplicationSourceTests extends OpenSearchIndexLevelRepli
         try {
             emptyIndexShard = newStartedShard(
                 true,
-                Settings.builder().put(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, true).build(),
+                settings,
                 new InternalEngineFactory()
             );
             RemoteSegmentStoreDirectory remoteSegmentStoreDirectory =

